src/Relude/Extra/Foldable1.hs

```diff
@@ -139,6 +155,22 @@ instance Foldable1 NonEmpty where
     {-# INLINE maximum1 #-}
     {-# INLINE minimum1 #-}
 
+    maximumOn1 :: Ord b => (a -> b) -> NonEmpty a -> a
+    maximumOn1 func = foldl1' (cmpOn func)
```

Comment:
I think you don't need to pass `func` as an argument to `cmpOn` since `func` is already in scope for `cmpOn`